src/api/foreground/foreground-setup-wallet-sdk.ts

```diff
@@ -12,102 +12,134 @@ export function setupWalletSDK(targetWindow: Window = window) {
   /** Init events */
   const events = mitt<InjectedEvents>();
 
-  /** Init wallet API */
-  const WalletAPI: Record<string, any> = {
-    walletName: "ArConnect",
-    walletVersion: version,
-    events
-  };
-
-  /** Inject each module */
+  /**
+   * Store modules in a map for quick lookup.
+   */
+  const moduleMap = new Map<string, (typeof foregroundModules)[number]>();
   for (const mod of foregroundModules) {
-    /** Handle foreground module and forward the result to the background */
-    WalletAPI[mod.functionName] = (...params: any[]) =>
```

Comment:
Did you check if theses changes affect the BE API?

Reply:
I was able to spin up local BE setup and was able to connect to dapps without issues